prov/sockets/src/sock_progress.c

```diff
@@ -2111,7 +2120,8 @@ int sock_pe_progress_rx_ctx(struct sock_pe *pe, struct sock_rx_ctx *rx_ctx)
 	struct dlist_entry *entry;
 	struct sock_pe_entry *pe_entry;
 
-	fastlock_acquire(&pe->lock);
+	if (fastlock_tryacquire(&pe->lock))
+		return 0;
```

Comment:
Yes, this is to avoid a 'possible' deadlock in manual progress mode.
This is a corner case.

In manual progress mode, cntr_read results in calling progress engine for each tx/rx context, which are maintained as a list in cntr. Calling progress engine results in acquiring pe-lock.
Now, another thread might have already been in progress engine and wants to acquire the cntr_lock. So, to avoid the "hold and wait", i added the try-lock for progress engine.

But, now I see another possible case like the following:
cntr_lock -> progress_engine_lock -> cntr_lock, (for the same cntr, and by same thread) because of cntr_read() -> progress() -> cntr_update(). 

Another solution is to release the cntr_lock before entering the progress engine.
(Same for CQ). But in this case, I need to make sure the tx/rx context list updates by another thread are not causing issues to the current thread's list traversal. The more I think, this seems to be a better solution, and avoids both the cases.